php/libraries/NDB_Page.class.inc

```diff
@@ -590,6 +590,257 @@ class NDB_Page
+        $tpl_data['commentID']   = $request['commentID'] ?? '';
+        $tpl_data['dynamictabs'] = $request['dynamictabs'] ?? '';
+
+        if (!empty($request['candID'])) {
+            $candidate = Candidate::singleton($request['candID']);
+
+            $tpl_data['candidate'] = $candidate->getData();
+        }
+
+        // Stuff that probably shouldn't be here, but exists because it was in
+        // main.php
+
+        // This seems to only be used in imaging_browser, it can probably be
+        // moved to properly use OOP.
+        $tpl_data['formaction'] = $this->FormAction ?? '';
+        // Doesn't appear to be used
+        $tpl_data['lastURL'] = $_SESSION['State']->getLastURL();
+        // I don't think anyone uses this. It's not really supported
+        $tpl_data['css'] = $config->getSetting('css');
+
+        // This just feels wrong
+        if (method_exists($this, 'getControlPanel')) {
+            $tpl_data['control_panel'] = $this->getControlPanel();
+        }
+        if (method_exists($this, 'getFeedbackPanel')
+            && $user->hasPermission('bvl_feedback')
+        ) {
+            $tpl_data['feedback_panel'] = $this->getFeedbackPanel(
+                $get['candID'],
+                $get['sessionID'] ?? null
+            );
+
+            $tpl_data['bvl_feedback'] = NDB_BVL_Feedback::bvlFeedbackPossible(
+                $this->page
+            );
+        }
+
+        // This shouldn't exist. (And if it does, it shouldn't reference
+        // mantis..)
+        $tpl_data['mantis_url'] = $config->getSetting('mantis_url');
+
+        // We're back in the territory of stuff that belongs here..
+
+        // Variables that get passed along to the LorisHelper javascript object.
+        $tpl_data['studyParams'] = array(
+                                    'useEDC'      => $config->getSetting('useEDC'),
+                                    'useProband'  => $config->getSetting(
+                                        'useProband'
+                                    ),
+                                    'useFamilyID' => $config->getSetting(
+                                        'useFamilyID'
+                                    ),
+                                    'useConsent'  => $config->getSetting(
+                                        'ConsentModule'
+                                    )['useConsent'],
+                                   );
+        $tpl_data['jsonParams']  = json_encode(
+            array(
+             'BaseURL'   => $tpl_data['baseurl'],
+             'TestName'  => $tpl_data['test_name'],
+             'Subtest'   => $tpl_data['subtest'],
+             'CandID'    => $tpl_data['candID'],
+             'SessionID' => $tpl_data['sessionID'],
+             'CommentID' => $tpl_data['commentID'],
+            )
+        );
+
+        // User related template variables that used to be in main.php.
+        $site_arr = $user->getData('CenterIDs');
```

Comment:
@driusan 
as discussed yesterday, please check PR #3272;
those lines in particular:
https://github.com/MounaSafiHarab/Loris/blob/7a57c374674536d1d56e8859ce421ab7d33c70be/htdocs/main.php#L85-L90